esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -0,0 +1,432 @@
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "ws_workunitsService.hpp"
+#include "ws_wudetails.hpp"
+#include "jlib.hpp"
+#include "workunit.hpp"
+#include "jset.hpp"
+#include "jstatcodes.h"
+
+class WUDetailsVisitor : public IWuScopeVisitor
+{
+public:
+    WUDetailsVisitor(IConstWUAttributeOptions & _attribOptions, IConstWUAttributeToReturn & _attribsToReturn);
+    virtual ~WUDetailsVisitor(){};
+    virtual void noteStatistic(StatisticKind kind, unsigned __int64 value, IConstWUStatistic & extra) override;
+    virtual void noteAttribute(WuAttr attr, const char * value) override;
+    virtual void noteHint(const char * kind, const char * value) override;
+
+    void resetScope();
+    IArrayOf<IEspWUResponseAttribute> & getCurrentAttribs() { return currentAttribs;}
+    unsigned __int64 getMaxTimestamp() const { return maxTimestamp;}
+    void setScopeType(const StatisticScopeType _sst) { currentStatisticScopeType = _sst; }
+
+private:
+    static const unsigned StatisticFilterMaskSize = StatisticKind::StMax;
+    static const unsigned AttributeFilterMaskSize = WuAttr::WAMax-WANone;
+
+    IConstWUAttributeOptions & attribOptions;
+    IConstWUAttributeToReturn & attribsToReturn;
+    bool onlyDynamic;
+    bool includeName;
+    bool includeRawValue;
+    bool includeFormatted;
+    bool includeMeasure;
+    bool includeCreator;
+    bool includeCreatorType;
+
+    bool returnStatisticListSpecified;
+    Owned<IBitSet> returnStatisticList;
+    bool returnAttributeSpecified;
+    Owned<IBitSet> returnAttributeList;
+    bool scopeTypeOverrideSpecified;
+    std::map<StatisticScopeType,std::vector<StatisticKind>> scopeTypeOverrideStatistic;
+    std::map<StatisticScopeType,std::vector<WuAttr>> scopeTypeOverrideAttribute;
+
+    unsigned __int64 maxTimestamp;
+    IArrayOf<IEspWUResponseAttribute> currentAttribs;
+    StatisticScopeType currentStatisticScopeType;
+
+    void buildAttribListToReturn(IConstWUAttributeToReturn & attribsToReturn);
+};
+
+WUDetailsVisitor::WUDetailsVisitor(IConstWUAttributeOptions & _attribOptions, IConstWUAttributeToReturn & _attribsToReturn)
+:  attribOptions(_attribOptions), attribsToReturn(_attribsToReturn), scopeTypeOverrideSpecified(false),
+   onlyDynamic(_attribsToReturn.getOnlyDynamic()), maxTimestamp(0), returnStatisticListSpecified(false),
+   returnAttributeSpecified(false), currentStatisticScopeType(SSTnone)
+{
+    if (*attribsToReturn.getMeasure())
+        onlyDynamic=true;
+    includeName = attribOptions.getIncludeName();
```

Comment:
if onlyDynamic is set then you should be calling addOutputProperties(PTstatistics);

Reply:
onlyDynamic no longer used. (Removed as new field added: AllStatistics, AllHints and AllAttributes. 
 These may fields may be used to achieve the results previously achieved with onlyDynamic)